evm/src/util.rs

```diff
@@ -161,12 +160,10 @@ pub(crate) fn le_limbs_to_biguint(x: &[u128]) -> BigUint {
     )
 }
 
-#[cfg(test)]
 pub(crate) fn mem_vec_to_biguint(x: &[U256]) -> BigUint {
     le_limbs_to_biguint(&x.iter().map(|&n| n.try_into().unwrap()).collect_vec())
```

Comment:
Another optimization opportunity here (doesn't need to be done in this PR, or TBH at all, but I'd add a note for whoever's looking at this code next): the iterator is collected into a vector, only for the `Vec` to be iterated over in `le_limbs_to_biguint`.